phoenix-core/src/main/java/org/apache/phoenix/iterate/RegionScannerFactory.java

```diff
@@ -169,7 +177,13 @@ public long getMvccReadPoint() {
       @Override
       public boolean nextRaw(List<Cell> result) throws IOException {
         try {
-          boolean next = s.nextRaw(result);
+          boolean next;
+          do {
```

Comment:
This little bit of loop code gets repeated in several places -- any way to encapsulate it somewhere that can be reused?

Reply:
Also some comments explaining what verifyRow == false or true means -- I had to read the verifyRow implementation a couple of times to figure it out. :-)

Reply:
Will do as you suggest it

Reply:
Done